etl/steps/data/meadow/hmd/2024-11-19/hfd.py

```diff
@@ -0,0 +1,349 @@
+        PATFR   PATFR1   PATFR2   PATFR3   PATFR4  PATFR5p
+
+    pmab
+        Period table mean ages at birth by birth order
+        TMAB    TMAB1    TMAB2    TMAB3    TMAB4   TMAB5p
+    pmabc
+        Period table mean ages at birth by birth order (based on parity distribution from population census)
+        TMAB    TMAB1    TMAB2    TMAB3    TMAB4   TMAB5p
+
+    sdmabRR
+        Standard deviation in period mean ages at birth and standard deviation in period mean ages at birth by age 40
+        sdMAB    sdMAB40
+    sdmabRRbo
+        Standard deviation in period mean ages at birth by birth order and standard deviation in period mean ages at birth by birth order by age 40
+        sdMAB      sdMAB1      sdMAB2      sdMAB3      sdMAB4     sdMAB5p     sdMAB40   sdMAB40_1   sdMAB40_2   sdMAB40_3   sdMAB40_4  sdMAB40_5p
+
+    tfrRR
+        Period total fertility rates and period total fertility rates by age 40
+        TFR     TFR40
+    tfrRRbo
+        Period total fertility rates by birth order and period total fertility rates by birth order by age 40
+        TFR      TFR1      TFR2      TFR3      TFR4     TFR5p     TFR40   TFR40_1   TFR40_2   TFR40_3   TFR40_4  TFR40_5p
+
+
+    => OUTPUT
+
+    columns
+        adjTFR
+        CBR
+        MAB
+        MAB40
+        sdMAB
+        sdMAB40
+        TFR
+        TFR40
+        PATFR
+        PATFR_c
+        TMAB
+        TMAB_c
+    dimensions
+        code, year, birth_order
+
+3y: code, year, age
+4y: code, year, age, cohort
+2c: code, cohort
+3c: code, cohort, age
+3x: code, cohort, x
+3X: code, year, x
+4A: code, year, cohort, ardy
+----
+
+3y asfrRR
+3y asfrRRbo
+4y asfrTR
+4y asfrTRbo
+3c asfrVH
+3c asfrVHbo
+4A asfrVV
+4A asfrVVbo
+
+3y birthsRR
+3y birthsRRbo
+4y birthsTR
+4y birthsTRbo
+3c birthsVH
+3c birthsVHbo
+4A birthsVV
+4A birthsVVbo
+
+
+3c ccfrVH
+3c ccfrVHbo
+
+3x cft
+
+3y cpfrRR
+3y cpfrRRbo
+4A cpfrVV
+4A cpfrVVbo
+
+3y exposRR
+3y exposRRpa
+3y exposRRpac
+4y exposTR
+3c exposVH
+4A exposVV
+
+2c mabVH
+2c mabVHbo
+
+3y mi
+3y mic
+
+3X pft
+3X pftc
+
+2c pprVHbo
+
+2c sdmabVH
+2c sdmabVHbo
+
+2c tfrVH
+2c tfrVHbo
+
+2y totbirthsRR
+2y totbirthsRRbo
+
+"""
+from pathlib import Path
+
+import owid.catalog.processing as pr
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Behavior 1:
+# Mosrtly use ["code", "year", "age"] except:
+# - When table end with 'TR' or 'TRbo': ["code", "year", "year", "cohort"] for those
+# - When table end with 'VV' or 'VVbo': ["code", "year", "ardy"] for those
+cols_1 = (
+    "asfr",
+    "births",
+    "ccfr",
+    "cpfr",
+    "expos",
+    "mi",
+)
+# Behaviour 2: Use ["code", "cohort"] always
+cols_2 = ("pprVHbo",)
+# Behavior 3: Use ["code", "year", "age"] always
+cols_3 = (
+    "cft",
+    "pft",
+)
+
+
+def get_cols_format(tb, short_name):
```

Comment:
It works fine, but wouldn't it be better to define a dictionary?

Reply:
yeah, i considered this. But somehow this made sense given the number of tables. I might revisit this though.